src/lib/markdown_generators.js

```diff
@@ -0,0 +1,184 @@
+      ),
+      department: titleCase.titleCase(
+        courseData["url"].split("/")[2].replace(/-/g, " ")
+      ),
+      topics:        courseData["course_collections"].map(makeTopic),
+      course_number: courseNumber,
+      term:          `${courseData["from_semester"]} ${courseData["from_year"]}`,
+      level:         courseData["course_level"],
+      menu:          {
+        main: {
+          weight: -10
+        }
+      }
+    }
+  }
+  return `---\n${yaml.safeDump(frontMatter)}---\n`
+}
+
+const generateCourseSectionFrontMatter = (title, menuIndex) => {
+  /*
+      Generate the front matter metadata for a course section given a title and menu index
+      */
+  return `---\n${yaml.safeDump({
+    title: title,
+    menu:  {
+      main: {
+        weight: menuIndex
+      }
+    }
+  })}---\n`
+}
+
+const generateCourseFeatures = courseData => {
+  /*
+      Generate markdown for the "Course Features" section of the home page
+      */
+  const courseFeaturesHeader = markdown.headers.hX(5, "Course Features")
+  const courseFeatures = courseData["course_features"].map(courseFeature => {
+    const urlParts = courseFeature["ocw_feature_url"]
+      .replace(/\/index.htm?l/, "/")
+      .split("/")
+    const url = urlParts[urlParts.length - 1]
+    return markdown.misc.link(courseFeature["ocw_feature"], url)
+  })
+  return `${courseFeaturesHeader}\n${markdown.lists.ul(courseFeatures)}`
+}
+
+const generateCourseCollections = courseData => {
+  /*
+      Generate markdown for the "Course Collections" section of the home page
+      */
+  const courseCollectionsHeader = markdown.headers.hX(5, "Course Collections")
+  const courseCollectionsSubHeader = `\nSee related courses in the following collections:\n\n${markdown.emphasis.i(
+    "Find Courses by Topic"
+  )}\n\n`
+  const courseCollections = courseData["course_collections"].map(
+    courseCollection => {
+      const feature = courseCollection["ocw_feature"]
+      const subfeature = courseCollection["ocw_subfeature"]
+      const specialty = courseCollection["ocw_specialty"]
+      let collection = feature
+      if (subfeature) {
+        collection = `${collection} > ${subfeature}`
+      }
+      if (specialty) {
+        collection = `${collection} > ${specialty}`
+      }
+      return markdown.misc.link(collection, "#")
+    }
+  )
+  return `${courseCollectionsHeader}${courseCollectionsSubHeader}${markdown.lists.ul(
+    courseCollections
+  )}`
+}
+
+const generateCourseSectionMarkdown = (page, courseData) => {
+  /*
+      Generate markdown a given course section page
+      */
+  let htmlStr = page["text"]
+  courseData["course_pages"].forEach(coursePage => {
+    const placeholder = `./resolveuid/${coursePage["uid"]}`
+    if (htmlStr.includes(placeholder)) {
+      htmlStr = htmlStr.replace(
+        placeholder,
+        `/sections/${coursePage["short_url"]}`
+      )
+    }
+  })
+  courseData["course_files"].forEach(media => {
```

Comment:
We should cut this part until we need it later, when we have resolved the TODO question below